lib/src/custom_image_crop_widget.dart

```diff
@@ -0,0 +1,249 @@
+import 'dart:async';
+import 'dart:math';
+import 'dart:ui' as ui;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:gesture_x_detector/gesture_x_detector.dart';
+import 'package:vector_math/vector_math_64.dart' as vector_math;
+
+import 'package:custom_image_crop/src/controller.dart';
+import 'package:custom_image_crop/src/dotted_path_painter.dart';
+import 'package:custom_image_crop/src/inverted_clipper.dart';
+import 'package:custom_image_crop/src/model.dart';
+
+class CustomImageCrop extends StatefulWidget {
+  final ImageProvider image;
+  final CustomImageCropController cropController;
+  final Color backgroundColor;
+  final Color overlayColor;
+  final Function(double) path;
```

Comment:
When using a path I get a strange clipped image

code:
```
            child: CustomImageCrop(
              cropController: controller,
              path: (cropWidth) {
                return Path()
                  ..moveTo(50, 50)
                  ..lineTo(100, 50)
                  ..lineTo(100, 100)
                  ..lineTo(50, 100)
                  ..close();
              },
              image: const AssetImage('assets/test.png'), // Any Imageprovider will work, try with a NetworkImage for example...
            ),
```


Image cropper with path             |  Cropped Image
:-------------------------:|:-------------------------:
![image](https://user-images.githubusercontent.com/21172855/112948117-f2f05e80-9137-11eb-916a-714aecfc1ef4.png)  |  ![image](https://user-images.githubusercontent.com/21172855/112948130-f7b51280-9137-11eb-9660-3175fe6ed01a.png)

Reply:
Yes I see why. The path is meant to be in regards to cropwidth, width and height. Using const ints like 100 and 50 they're not scaled during the crop to the new width and height, instead the path is being called again and expects to be scaled automatically with the new width and height during the crop 😕 
I could try and scale the path instead or we can add a comment that mentions to always use width and height

Reply:
removed path for now